### ISSUE TITLE
Single-char lookaround?

### ISSUE THREAD

**Issue by CAD97** [rust-lang/regex#833]
TL;DR: Is it reasonable to support lookaround (`(?=`, `(?!`, `(?<=`, `(?<!`) where the lookahead is restricted to a single character class?

#### Describe your feature request

Fundamentally, it seems that the functionality already exists, because `\b` (word boundary) exists. `\b` is a kind of lookaround -- it asserts that one side is `\w` and the other side is `\W|\A|\z` (anchors can't be put in a character class). Expanded to explicit lookaround, you could write `\b` as `(?:(?<=\w)(?!\w)|(?<!\w)(?=\w))`.

It would be nice to be able to do what `\b` does, but with arbitrary character classes, not just `\w`. The obvious example is reserved words in a language; given UAX31 identifiers `\p{XID_START}\p{XID_CONTINUE}*`, you could match a reserved word as `\bbool\b`, but this is not quite correct, because `\w` which `\b` is checking for is `[\p{Alphabetic}\p{M}\p{Nd}\p{Pc}\p{Join_Control}]`, which is a different character class than `\p{XID_CONTINUE}`. The correct way to match a reserved word would instead be `(?<!\p{XID_CONTINUE})bool(?!\p{XID_CONTINUE})`, which properly matches borders of the UAX31 identifier syntax.

Obviously, even if much of `\b`'s handling can be reused, this is a relatively major expansion to what regex is capable of (e.g. regex-syntax doesn't even parse lookaround groups). However, it seemed worth asking, if only to find out why it's more complicated than it seems to do so.

**Comment by BurntSushi** [comment]
I am pretty opposed to this personally. Dealing with a Unicode-aware `\b` is a huuuuuge pain, and is generally treated as a special case in most places. Adding more support for Unicode-aware look-around is not something I'm personally interested in pursuing or maintaining at this point.

**Comment by BurntSushi** [comment]
I'm going to close this out because I personally don't see this happening.
